placeholder-transaction-trait/src/transaction_trait.rs

```diff
@@ -0,0 +1,57 @@
+use {
+    crate::MessageTrait,
+    solana_sdk::{
+        hash::Hash,
+        packet::Packet,
+        signature::Signature,
+        transaction::{TransactionAccountLocks, TransactionError, VersionedTransaction},
+    },
+};
+
+mod reference;
+mod sanitized_transaction;
+
+pub trait TransactionTrait: MessageTrait {
+    /// Get the first signature of the message.
+    fn signature(&self) -> &Signature;
+
+    /// Get all the signatures of the message.
+    fn signatures(&self) -> &[Signature];
+
+    /// Returns the message hash.
+    // TODO: consider moving this to Message
+    fn message_hash(&self) -> &Hash;
```

Comment:
for `SanitizedTransaction` this just fetches a pre-computed message hash, which is why i'd put it here.
That's a current implementation detail and I feel doesn't make **too** much sense honestly.

Could also override the impl for SanitizedTransaction to get the cached value, while calculating it on SanitizedMessage and then move this to the MessageTrait